Fix question listing and navigation in quiz reports

rtibbles opened the pull request:
## Summary
* Annotate sections with start and end section numbers.
* Simplify attempt log listing by requiring sections, so only having a mobile and non-mobile condition
* Fix attempt log listing by including the section context in the question numbers.
* Remove background to show selection state.
* Update question number display to show the number globally for the quiz, rather than per section.
* Update learner quiz display to be correlated.
* Cleanup cruft in learner quiz display that was making our computed props over complicated and redundant.

## References
Fixes [#12334](https://github.com/learningequality/kolibri/issues/12334)

## Reviewer guidance
* Test multi-section quiz reports to ensure that selection state is properly shown.
* Test to ensure that navigation between items occurs properly.
* Regression test practice quiz reports for learners and coaches
* Regression test exercise reports for coaches
* Ensure that multi-section quiz engagement and question selection functions properly.

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests


## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

rtibbles committed:
Annotate sections with start and end section numbers.
Simplify and fix attempt log listing.
Remove background to show selection state.

pcenov commented:
Hi @rtibbles, everything seems ok but noticed 2 issues that can potentially cause confusion:

1. When all the sections are collapsed there's no indication to which section the displayed question belongs so that the learner/coach might have to expand several sections before finding out. Perhaps there should be a selection effect to the section title as well:

![2024-06-26_16-22-59](https://github.com/learningequality/kolibri/assets/79847249/946a6c72-c4e5-4926-aa6d-bc1cfadedc0c)

2. There's a mismatch between the question title within the section panel and the question title as displayed when viewing the details of the question. For example when I select Question 4 from the second section, it's displayed as Question 1 to the right and that's so for every new section:

https://github.com/learningequality/kolibri/assets/79847249/ad695713-7f6a-4aa5-8c27-0f402ae69c77

rtibbles commented:
Thanks @pcenov - definitely useful to add an indicator at the section level too - I will use the same background colour.

Will make sure that the titles match!

rtibbles committed:
Add section title and descriptions to reports.
Have global question numbers in reports.
Add section descriptions if they exist.

rtibbles committed:
Update learner quiz display to have consistent question numbering with reports.
Cleanup unnecessary props, computed props, and vuex getters.

rtibbles commented:
Hi @pcenov I have fixed both these issues, and moved to a consistent 'global across the quiz' question numbering.

To be completely consistent, I have also updated this for learners taking the quiz too, so additional testing there would be very useful.

pcenov reviewed (approved):
Thanks @rtibbles LGTM!

marcellamaki commented on kolibri/core/assets/src/exams/utils.js:
```diff
@@ -254,3 +254,26 @@ export function getExamReport(examId, tryIndex = 0, questionNumber = 0, interact
+    const annotatedSection = {
+      ...section,
+      startQuestionNumber,
+      endQuestionNumber: startQuestionNumber + section.questions.length - 1,
```
non-blocking but I think adding a comment here about the numbering convention would make this section a bit more readable. (I realize it's a refactor from a different file)

marcellamaki reviewed (approved):
Holding the context for all of this is a bit of a challenge, but that's a factor of the complexity of some of this code. One non-blocking suggestion for an additional comment

rtibbles committed:
Add explanatory comment for annotation.
